pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -961,21 +1094,42 @@ func afwSchedulerEntry(client splcommon.ControllerClient, cr splcommon.MetaObjec
 	}
 	scopedLog.Info("List of cluster scoped apps(appName:digest) for this reconcile entry", "apps", clusterAppsList)
 
+	targetPodName := getApplicablePodNameForWorker(cr, 0)
+
 	// Wait for the yield function to finish
 	afwPipeline.phaseWaiter.Add(1)
 	go func(afwEntryTime int64) {
 		for {
-			if afwEntryTime+maxRunTimeBeforeAttemptingYield < time.Now().Unix() || afwPipeline.isPipelineEmpty() {
+			if afwEntryTime+maxRunTimeBeforeAttemptingYield < time.Now().Unix() || (afwPipeline.isPipelineEmpty() && getBundlePushState(afwPipeline) != enterpriseApi.BundlePushPending) {
 				scopedLog.Info("Yielding from AFW scheduler", "time elapsed", time.Now().Unix()-afwEntryTime)
 
 				// Trigger termination by closing the channel
 				close(afwPipeline.sigTerm)
 				afwPipeline.phaseWaiter.Done()
 				break
 			} else {
-				if len(clusterScopedApps) > 0 && appDeployContext.BundlePushStatus.BudlePushStage < enterpriseApi.BundlePushPending {
-					if checkIfBundlePushNeeded(clusterScopedApps) {
-						// Trigger the bundle push playbook: CSPL-1332, CSPL-1333
+				if len(clusterScopedApps) > 0 && getBundlePushState(afwPipeline) != enterpriseApi.BundlePushComplete && checkIfBundlePushNeeded(clusterScopedApps) {
+					// Trigger the bundle push playbook: CSPL-1332, CSPL-1333
+					kind := cr.GetObjectKind().GroupVersionKind().Kind
+
+					if afwPipeline.playBookContext == nil {
+						afwPipeline.playBookContext = getPlayBookContext(client, cr, afwPipeline, targetPodName, kind)
+					}
+					// run the playbook to issue bundle push command on the pod
+					err = afwPipeline.playBookContext.runPlayBook()
+					if err != nil {
+						// If the error is due to a bundle which is already present, don't do anything.
+						// In the next reconcile we will mark it as bundle push complete
+						if strings.Contains(err.Error(), idxcBundleAlreadyPresentStr) {
+							continue
```

Comment:
nit: the continue will miss the sleep, but hope it's ok for now, as this code move to the install handler.